### DISCUSSION TITLE
画像クリック時の拡大動作(lightbox)を動作させないオプションの追加

### DISCUSSION THREAD

**Discussion by newspot4034** [growilabs/growi#9030] [Ideas / Requests]
いつも大変お世話になっております。
職場でナレッジ共有のためのWikiをGrowiで運用しております。
利用者からいくつか改善要望が上がったものがあり、以下リクエストさせてもらえればと思います。

# 環境
- Growi 6.3.4

# 現在の動作
- 画像ファイルをクリックした際に、周囲が暗転し画像のみが拡大された状態となる

# 機能追加リクエスト
- 画像クリック時の拡大動作(lightbox)を動作させないオプションの追加
- 以前のバージョン同様に画像がクリック不可となるのが望ましい

# 追加要望の経緯
トップページにおいてaタグとimgタグを併用し、画像リンクとして
他ページに飛ぶ方法が視覚的にわかりやすくなるようにしているが
現状の動作だと「画像クリック → ページ読み込みまでの間一瞬暗転 → ページ読み込み」といった
動きとなっており特段影響はないものの視覚的に非常に気になる状態となっている。

元々v5系列などでは存在しない機能だったので、OFF/ONのフラグを実装いただき
以前のような拡大不可となる選択肢も実装いただけると助かります。

**Comment by goofmint** [comment]
@newspot4034 横からすみません。

v6.3が手元にないので検証できていないのですが、該当する画像に `pointer-events: none;` を設定すると拡大動作（クリック動作）が無効になります。

v7系であれば、 `.wiki` 以下がコンテンツになるので、以下のようなカスタムCSSで画像の拡大表示が無効になりました（管理画面で設定）。

```css
.wiki img {
  pointer-events: none;
}
```

v6.3系でのHTML構造が把握できていませんが、似たような形でできるかな…と。

**Reply by newspot4034** [reply]
情報ありがとうございます。
v6.3で確認しましたが、通常の画像のクリック判定は期待通り無効化することができました。

ただ、画像リンクにしている画像についてはクリック判定が極小化してしまい、
画像部分の中央部しか反応しないような状況となってしまったため切り戻しする次第となりました。
せっかく情報いただきましたのに申し訳ございません。

imgタグで挿入した画像についてはlightboxが動作しない、といったようなものがあれば
非常に助かるのですが同じような事象に困っている方は比較的少数派なのでしょうか・・・

**Comment by goofmint** [comment]
画像リンクというと、 `<a><img /></a>` のような構成ということでしょうか？

以下のCSSの場合、aタグを使っていない場合のみ、クリックイベントが無効になるかと思います（またしてもv7の場合ですが…）。ただし、リンクは別タブで開きつつ、拡大表示もされてしまいます…。

```css
.wiki p > img {
  pointer-events: none;
}
```

実際の動作はこんな感じになります。

```markdown
// 拡大表示せず
![画像のalt](/attachment/669a5c8b8d45e62b521d0702)

// Yahooへのリンクは有効
[![画像のalt](/attachment/669a5c8b8d45e62b521d0702)](https://www.yahoo.co.jp)
```
